app/Http/Controllers/ProductController.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\common;
+use App\Models\Product;
+use App\Rules\RangeRule;
+use Illuminate\Http\Request;
+
+class ProductController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     */
+    use common;
+
+    public function index()
+    {
+        //all products
+        $products=Product::orderBy('created_at','desc')->get();
```

Comment:
there is a built in function called `latest()` do the same job 

look at its internal implementation
```php
public function latest($column = 'created_at')
    {
        return $this->orderBy($column, 'desc');
    }
```

if you want to see its implementation go to `vendor/laravel/src/Illuminate/Database/Query/Builder.php`